gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcHeaderInterceptor.java

```diff
@@ -48,14 +49,25 @@
  */
 class GrpcHeaderInterceptor implements ClientInterceptor {
   private final Map<Metadata.Key<String>, String> staticHeaders;
+  private final String userAgentHeader;
 
   public GrpcHeaderInterceptor(Map<String, String> staticHeaders) {
     ImmutableMap.Builder<Metadata.Key<String>, String> grpcHeaders = ImmutableMap.builder();
+    String userAgentStaticHeader = null;
     for (Map.Entry<String, String> header : staticHeaders.entrySet()) {
-      grpcHeaders.put(
-          Metadata.Key.of(header.getKey(), Metadata.ASCII_STRING_MARSHALLER), header.getValue());
+      Metadata.Key<String> headerKey =
+          Metadata.Key.of(header.getKey(), Metadata.ASCII_STRING_MARSHALLER);
+
+      // User-Agent is overridden on gRPC level. The custom User-Agent is supposed to be provided
+      // differently and only merging with gRPC default value for User-Agent is permitted.
```

Comment:
Do you have a reference for this?

Reply:
What do you mean by reference?

Reply:
I mean a link to documentation that talks about this.

Reply:
I don't know about documentation regarding this. The comment above is based on what I've found from debugging grpc code.